zaza/openstack/charm_tests/kerberos/setup.py

```diff
@@ -0,0 +1,239 @@
+    demo_user = keystone_client.users.create(kerberos_user,
+                                             domain=domain,
+                                             project=project,
+                                             password=kerberos_password,
+                                             email='demo@demo.com',
+                                             description='Demo User',
+                                             enabled=True)
+    admin_role = keystone_client.roles.find(name=role)
+    keystone_client.roles.grant(
+        admin_role,
+        user=demo_user,
+        project_domain=domain,
+        project=project
+    )
+    keystone_client.roles.grant(
+        admin_role,
+        user=demo_user,
+        domain=domain
+    )
+
+
+def setup_kerberos_configuration_for_test_host():
+    """Retrieve the keytab and krb5.conf to setup the ubuntu test host."""
+    kerberos_server = zaza.model.get_units('kerberos-server')[0]
+    ubuntu_test_host = zaza.model.get_units('ubuntu-test-host')[0]
+
+    dump_file = "krb5.keytab"
+    remote_file = "/etc/krb5.keytab"
+    host_keytab_path = '/home/ubuntu/krb5.keytab'
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmp_file = "{}/{}".format(tmpdirname, dump_file)
+        logging.info("Retrieving {} from {}.".format(remote_file,
+                                                     kerberos_server.name))
+        zaza.model.scp_from_unit(
+            kerberos_server.name,
+            remote_file,
+            tmp_file)
+
+        logging.info("SCP {} to {} on {}.".format(tmp_file,
+                                                  host_keytab_path,
+                                                  ubuntu_test_host.name))
+        zaza.model.scp_to_unit(
+            ubuntu_test_host.name,
+            tmp_file,
+            host_keytab_path)
+
+    dump_file = "krb5.conf"
+    remote_file = "/etc/krb5.conf"
+    temp_krb5_path = "/home/ubuntu/krb5.conf"
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmp_file = "{}/{}".format(tmpdirname, dump_file)
+        logging.info("Retrieving {} from {}".format(remote_file,
+                                                    kerberos_server.name))
+        zaza.model.scp_from_unit(
+            kerberos_server.name,
+            remote_file,
+            tmp_file)
+
+        logging.info("SCP {} to {} on {}.".format(tmp_file,
+                                                  temp_krb5_path,
+                                                  ubuntu_test_host))
+        zaza.model.scp_to_unit(
+            ubuntu_test_host.name,
+            tmp_file,
+            temp_krb5_path)
+        logging.info('Moving {} to {} on {}.'.format(temp_krb5_path,
+                     remote_file, ubuntu_test_host.name))
+        zaza.model.run_on_unit(ubuntu_test_host.name, ('sudo mv {} {}'.
+                               format(temp_krb5_path, remote_file)))
+
+
+def install_apt_packages_on_ubuntu_test_host():
```

Comment:
Per 1:1 discussion, we always initialize OpenStack clients pythonically so that it tracks the same dependencies as defined in the requirements and test-requirements.

In this case, the client libraries are not coming from there, so we need to take note that of the risk.  As-is, the client package versions will always be aligned to the series of the cs:ubuntu unit, instead of being defined by the requirements.txt files.  At some point, this is nearly certain to break, as things are not always forward-compatible.

The best way to mitigate that risk, if there is no way to use the already-maintained requirements/test-requirements files, is to enable the Ubuntu Cloud Archive which aligns with the OpenStack version under test.

I'm not going to block on this now, but we need to log this as a bug for tech-debt to pay off here.